lib/backgroundscanner.php

```diff
@@ -84,21 +100,27 @@ public function run(){
 				->andWhere(
 					$qb->expr()->neq('fc.size', '0')
 				)
-				->setMaxResults(5)
+				->setMaxResults(self::BATCH_SIZE)
 			;
 			$result = $qb->execute();
 		} catch(\Exception $e) {
 			\OC::$server->getLogger()->error( __METHOD__ . ', exception: ' . $e->getMessage(), ['app' => 'files_antivirus']);
 			return;
 		}
 
-
-		$view = new \OC\Files\View('');
 		try {
 			while ($row = $result->fetch()) {
-				$path = $view->getPath($row['fileid']);
+				$fileId = $row['fileid'];
+				$owner = $this->getOwner($fileId);
```

Comment:
Are you excluding federated shares here ? Just asking because today I noticed that it would get a non-resolvable owner (the federated id of the remote user).